runtime/src/main/java/org/corfudb/runtime/collections/TransactionPoller.java

```diff
@@ -34,17 +34,30 @@
      */
     private final List<StreamingSubscriptionContext> streamContexts;
 
+    /**
+     * A reference to the transaction stream
+     */
+    private final IStreamView txnStream;
+
     /**
      * Constructor.
      *
      * @param streams The list of StreamingSubscriptionContexts to process.
      */
     public TransactionPoller(@Nonnull CorfuRuntime runtime,
-            @Nonnull List<StreamingSubscriptionContext> streams) {
+                             @Nonnull List<StreamingSubscriptionContext> streams) {
         this.runtime = runtime;
         this.streamContexts = streams.stream()
                 .sorted(Comparator.comparingLong(sc -> sc.getLastReadAddress()))
                 .collect(Collectors.toList());
+
+        StreamOptions options = StreamOptions.builder()
+                .ignoreTrimmed(true)
```

Comment:
Hmm, @sneginhal why is this true ?

Reply:
We need to define a generic API to handle TrimmedException for the caller.

Reply:
Trimmed exceptions for objects are automatically handled, but for raw streams usually not much can be done because the user expects a delta, but the delta has been trimmed. I don't see how we can automatically handle these exception because a trimmed exception indicates a loss of deltas, so the consumer has to first observe this exception, then do remedial actions related to their business logic and then re-seek and track deltas from the the current point in time.

Reply:
I can try to answer on behalf of @sneginhal  - this call is made by an internal transaction poller that is common to all the streams. If a stream tries to open something which is trimmed that error will be handled separately. We should not let the transaction poller itself fail with trimmed exception and simply proceed with the latest point.

Reply:
On a trim exception, the stream pointer has to be re-adjusted, but it doesn't seem like the logic is doing that.

Reply:
There is no ways for you to detect omissions if youe ignoe `TrimmedException`. How do you detect gaps in the stream?

Reply:
Can we resolve this ? I am not sure if I completely understand this and would want to make sure the correct behaviour is implemented.

Reply:
I have fixed the ignoreTrimmed exception in the latest patch 
https://github.com/CorfuDB/CorfuDB/pull/2287
thanks

Reply:
@hisundar please squash this in your PR. I'm just going to close this.